powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/SwitchConverter.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.powerfactory.converter;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.SwitchKind;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.powerfactory.converter.PowerFactoryImporter.ImportContext;
+import com.powsybl.powerfactory.converter.PowerFactoryImporter.NodeRef;
+import com.powsybl.powerfactory.model.DataObject;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+
+class SwitchConverter extends AbstractConverter {
+
+    SwitchConverter(ImportContext importContext, Network network) {
+        super(importContext, network);
+    }
+
+    void createFromElmCoup(DataObject elmCoup) {
+        List<NodeRef> nodeRefs = findNodes(elmCoup);
+        if (nodeRefs.size() != 2) {
+            LOGGER.warn("ElemCoup discarded as it does not have two ends '{}'", elmCoup);
+            return;
+        }
+        NodeRef nodeRef1 = nodeRefs.get(0);
+        NodeRef nodeRef2 = nodeRefs.get(1);
+        if (!nodeRef1.voltageLevelId.equals(nodeRef2.voltageLevelId)) {
+            throw new PowsyblException("ElmCoup not connected to same ElmSubstat at both sides: " + elmCoup);
+        }
+
+        VoltageLevel vl = getNetwork().getVoltageLevel(nodeRef1.voltageLevelId);
+        SwitchModel switchModel = SwitchModel.create(vl.getId(), elmCoup);
```

Comment:
avoid code duplication by calling `createSwitch(vl, nodeRef1.node, nodeRef2.node, elmCoup)`. 

`createFromStaSwitch(...)` should be renamed to `createSwitch(...)`. Its last parameter should be named `dataObject` instead of `staSwitch`.

Reply:
Done